glean-core/src/metrics/experiment.rs

```diff
@@ -101,23 +117,38 @@ impl ExperimentMetric {
         }
 
         // Make sure that branch id is within the expected limit.
-        let truncated_branch = if branch.len() > MAX_EXPERIMENTS_IDS_LEN {
+        let truncated_branch = if branch.len() > MAX_EXPERIMENTS_BRANCH_LEN {
             log::warn!(
                 "Value length {} for branch exceeds maximum of {}",
                 branch.len(),
-                MAX_EXPERIMENTS_IDS_LEN
+                MAX_EXPERIMENTS_BRANCH_LEN
             );
-            truncate_string_at_boundary(branch, MAX_EXPERIMENTS_IDS_LEN)
+            truncate_string_at_boundary(branch, MAX_EXPERIMENTS_BRANCH_LEN)
         } else {
             branch
         };
 
-        // TODO (bug 1566001): add sane limits for the extra keys and
-        // values as well.
+        // Apply limits to extras
+        let truncated_extras = if extra.is_some() {
```

Comment:
You could do:

```
let truncated_extras = extra.and_then(|extra| { ... })
```

to avoid the `extra.unwrap()` and `else { None }`.

Maybe just a personal preference because I like to avoid calling `unwrap()` in non-test code since it *could* panic (even though in this specific case it never would).

But defer to the real Rust expert @badboy on this one.

Reply:
TIL: This sounds like a better way to handle the Optional value rather than by force unwrapping it with `unwrap`.  Chalk this up to my lack of Rust knowledge.